# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/nsContentSecurityManager.cpp
# Commit: 74f99033251c
# Full Hash: 74f99033251cd6c36eb6b0b5a4713eb6b5c793fa
# Author: Jonathan Kingston <jkt@mozilla.com>
# Date: 2019-02-13 09:48:26
# Regressor Bug: 1520868
# File Overlap Count: 1
# Description:
#   Bug 1520868 - Replacing AsyncOpen2 with AsyncOpen always r=valentin
#   
#   Replacing js and text occurences of asyncOpen2
#   Replacing open2 with open
#   
# ==============================================================================

diff -r 3378fd5f4f22 -r 74f99033251c dom/security/nsContentSecurityManager.cpp
--- a/dom/security/nsContentSecurityManager.cpp	Tue Feb 12 17:58:19 2019 +0100
+++ b/dom/security/nsContentSecurityManager.cpp	Tue Feb 12 16:08:25 2019 +0000
@@ -749,9 +749,6 @@
     MOZ_LOG(sCSMLog, LogLevel::Debug,
             ("  initalSecurityChecksDone: %s\n",
              aLoadInfo->GetInitialSecurityCheckDone() ? "true" : "false"));
-    MOZ_LOG(sCSMLog, LogLevel::Debug,
-            ("  enforceSecurity: %s\n",
-             aLoadInfo->GetEnforceSecurity() ? "true" : "false"));
 
     // Log CSPrequestPrincipal
     nsCOMPtr<nsIContentSecurityPolicy> csp;
@@ -789,7 +786,7 @@
  * @param aChannel
  *    The channel to perform the security checks on.
  * @param aInAndOutListener
- *    The streamListener that is passed to channel->AsyncOpen2() that is now
+ *    The streamListener that is passed to channel->AsyncOpen() that is now
  * potentially wrappend within nsCORSListenerProxy() and becomes the
  * corsListener that now needs to be set as new streamListener on the channel.
  */
@@ -819,13 +816,6 @@
   nsresult rv = ValidateSecurityFlags(loadInfo);
   NS_ENSURE_SUCCESS(rv, rv);
 
-  // since aChannel was openend using asyncOpen2() we have to make sure
-  // that redirects of that channel also get openend using asyncOpen2()
-  // please note that some implementations of ::AsyncOpen2 might already
-  // have set that flag to true (e.g. nsViewSourceChannel) in which case
-  // we just set the flag again.
-  loadInfo->SetEnforceSecurity(true);
-
   if (loadInfo->GetSecurityMode() ==
       nsILoadInfo::SEC_REQUIRE_CORS_DATA_INHERITS) {
     rv = DoCORSChecks(aChannel, loadInfo, aInAndOutListener);
@@ -855,8 +845,7 @@
     nsIChannel* aOldChannel, nsIChannel* aNewChannel, uint32_t aRedirFlags,
     nsIAsyncVerifyRedirectCallback* aCb) {
   nsCOMPtr<nsILoadInfo> loadInfo = aOldChannel->GetLoadInfo();
-  // Are we enforcing security using LoadInfo?
-  if (loadInfo && loadInfo->GetEnforceSecurity()) {
+  if (loadInfo) {
     nsresult rv = CheckChannel(aNewChannel);
     if (NS_SUCCEEDED(rv)) {
       rv = CheckFTPSubresourceLoad(aNewChannel);